### PR TITLE
Add Touchstone to Auth Strategies

### LINKED ISSUE #4: Add SAML auth for login
# User Story

- As a stakeholder for MIT Open, I want to be able to access the project documentation with a familiar login
- As a product owner for MIT Open, I want to reduce friction for stakeholders accessing project documentation

# Description/Context
The NYTimes Library app uses the [Passport.JS](http://www.passportjs.org/) library to manage authentication. We would like to update this customization repo to use the [passport-saml](https://github.com/node-saml/passport-saml) strategy so that we can use Touchstone login for this documentation site. There is an example repository that uses that library that we can use as a reference [here](https://github.com/gbraad/passport-saml-example/)

# Acceptance Criteria

- [x] The site at https://mitopen-docs.odl.mit.edu uses Touchstone SAML as the login method

### PR DESCRIPTION
**Pull request by JenniWhitman**
# What are the relevant tickets?
Closes #4 

# Description (What does it do?)
Added an auth strategy for SAML to support Touchstone. Modified authenticate method and email to take into account the addition of a new strategy. This also supports the creation of the metadata page within the application.

# How can this be tested?
Sar has contacted IS&T in order to get the credentials created and registered as well as set up a QA environment for us to test this in.

A user should be able to click the touchstone login button and be logged in as their touchstone user successfully.

## Checklist:
- [ ] Get touchstone credentials from IS&T
- [ ] Create env variables to support this
- [ ] Test on QA

### CODE REVIEW

**Comment by shaidar** [comment]
👍

File: userAuth.js [hunk 1]
```diff
@@ -81,6 +103,18 @@ router.get('/auth/redirect', passport.authenticate(authStrategy, {failureRedirec
   res.redirect(req.session.authRedirect || formatUrl('/'))
 })
 
+router.get('/metadata', function (req, res) {
+  if (isTouchstoneSamlAuth) {
+    res.type('application/xml')
+    res.send((touchstoneSamlStrategy.generateServiceProviderMetadata(
+      process.env.TOUCHSTONE_SAML_MD_DECRYPTION_CERT,
```
Comment (shaidar) [inline userAuth.js:110]: What is this decrypting?

Comment (JenniWhitman) [reply userAuth.js:110]: Mike mentioned that his also used decryption cert which you're ensured & verify a secure transaction between IDP and service.  https://mitodl.slack.com/archives/G5RHT8GDD/p1690560282038369?thread_ts=1690556534.378059&cid=G5RHT8GDD for reference, here's Mike's mention. If we find it's unnecessary I can take it and the related PK out, but if there's a decrypt PK, the xml is expected to carry it as well.

also, from docs: https://github.com/node-saml/passport-saml#generateserviceprovidermetadata-decryptioncert-signingcert-

Comment (shaidar) [reply userAuth.js:110]: Yeah I get that, but guess my question was more around the wording itself. That's the cert that the IDP uses from the SP to encrypt the assertion and then the SP uses the private key to decrypt. Should we call it `TOUCHSTONE_SAML_SP_ENCRYPTION_CERT` instead?

**Pull request merged**